netket/machine/jax.py

```diff
@@ -149,14 +149,20 @@ def vector_jacobian_prod(
                 out = tree_map(jax.numpy.conjugate, pout[0])
 
             return out
+
         else:
+
+            vals, f_jvp = jax.vjp(
+                self._forward_fn, self._params, x.reshape((-1, x.shape[-1]))
+            )
+
+            pout = f_jvp(vec.reshape(vals.shape).conjugate())
+
             if conjugate and self._dtype is complex:
-                prodj = lambda j: jax.np.tensordot(vec, j.conjugate(), axes=vec.ndim)
```

Comment:
I put these because I didn't want to call f_jvp here explicitly, when the jacobian is passed to the function (otherwise you do more work). It'd be good to revert the changes to this function

Reply:
Okay true, I can write it this way